demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -336,10 +336,17 @@ def run_validations_on_file(self, file_path, pack_error_ignore_list, is_modified
             return True
 
         # id_set validation
-        if self.validate_in_id_set:
+        if self.id_set_validations:
+            # if True:
             click.echo(f"Validating id set registration for {file_path}")
             if not self.id_set_validator.is_file_valid_in_set(file_path):
                 return False
+            if file_type == FileType.INCIDENT_TYPE:
+                if not self.id_set_validator.is_incident_type_using_real_playbook(file_path):
+                    return False
+            if file_type == FileType.SCRIPT:
+                if not self.id_set_validator.is_script_using_real_command(file_path):
+                    return False
```

Comment:
I think these would fit better in the `id_set_validator.is_file_valid_in_set` method - you can pass the file_type there to save up time.